docs/quickstart.rst

```diff
@@ -491,141 +443,43 @@ The shorter syntax for calling methods is::
 
 The major shortcoming with this method is that expressing complex fields with many nested sub-structs can become very tedious.
 
-Once you have a promise, there are 2 ways of getting to the result. The first is to wait for it::
-
-    result = eval_promise.wait()
-
-The second is to build a promise chain by calling `then`::
+The returned promise can be handled like any other asyncio promise::
 
-    def do_stuff(val):
-        ...
-
-    eval_promise.then(do_stuff).wait()
+    result = await eval_promise()
 
 
 Pipelining
 ##########
-If a method returns values that are themselves capabilites, then you can access these fields before having to call `wait`. Doing this is called pipelining, and it allows Cap'n Proto to chain the calls without a round-trip occurring to the server::
+If a method returns values that are themselves capabilites, then you can access these fields before awaiting the promise. Doing this is called pipelining, and it allows Cap'n Proto to chain the calls without a round-trip occurring to the server::
 
     # evaluate returns `value` which is itself an interface.
     # You can call a new method on `value` without having to call wait first
     read_promise = eval_promise.value.read()
-    read_result = read_promise.wait() # only 1 wait call
-
-You can also chain promises with `then` and the same pipelining will occur::
-
-    read_result = eval_promise.then(lambda ret: ret.value.read()).wait()
-
+    read_result = await read_promise # only 1 await call
 
 Server
 ~~~~~~
-There are two ways to start a server: libcapnp socket wrapper and asyncio.
-The wrapper is easier to implement but is very limited (doesn't support SSL/TLS with Python).
-asyncio requires more setup and can be harder to debug; however, it does support SSL/TLS and has more control over the socket error conditions. asyncio also helps get around the threading limitations around the current pycapnp implementation has with libcapnp (pycapnp objects and functions must all be in the same thread). The asyncio Server is a bit more work to implement than an asyncio client as more error handling is required to deal with client connection/disconnection/timeout events.
-
 
 Starting a Server
-#################
-
-To start a server::
-
-    server = capnp.TwoPartyServer('*:60000', bootstrap=CalculatorImpl())
-    server.run_forever()
-
-.. note:: You can also pass a raw socket with a `fileno()` method to TwoPartyServer. In that case, `run_forever` will not work, and you will have to use `on_disconnect.wait()`.
-.. note:: This will not work with SSL/TLS, please see :ref:`rpc-asyncio-server`
+###########################
 
+Just like the asyncio client uses a asyncio client, the server uses a 
+asyncio server that can be created wit `capnp.AsyncIoStream.create_server`.
 
-.. _rpc-asyncio-server:
+We need an additional `new_connection()` method in the `Server` class to handle each of the incoming socket connections::
```

Comment:
Which `Server` class? Normally I'd think that this is a standalone function.

Reply:
🙈 no server class. It was a copy-paste mistake. I anyway rewrote this part now to explain the purpose better